opensafely/pull.py

```diff
@@ -119,3 +134,59 @@ def remove_deprecated_images(local_images):
             tag = f"{registry}/{image}"
             if tag in local_images:
                 subprocess.run(["docker", "image", "rm", tag], capture_output=True)
+
+
+session = requests.Session()
+token = None
+
+
+def get_remote_sha(full_name, tag):
+    """Get the current sha for a tag from a docker registry."""
+    global token
```

Comment:
Do we need to declare `token` global when it's already been defined outside of the function (line 140)?  It's already a global variable, isn't it?

Reply:
We do if I want to write to it, which we do if it's not set.

Otherwise, doing `token = ...` will only set a local `token` variable, not update the global one.